pkg/sync/sync.go

```diff
@@ -61,10 +87,12 @@ func checkFileExist(fileName string) bool {
 
 // makeTar function is copied from https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/cmd/cp.go#L309
 // srcPath is ignored if files is set
-func makeTar(srcPath, destPath string, writer io.Writer, files []string, globExps []string) error {
+// makeTar returns a pointer to the tar.Writer, and doesn't close it during this function. The close is called outside
+// because in the case for copying source files, additional files are required to be written to it after this function has
+// already returned - meaning we can no longer defer the Close.
+func makeTar(srcPath, destPath string, writer io.Writer, files []string, globExps []string) (*tar.Writer, error) {
 	// TODO: use compression here?
-	tarWriter := taro.NewWriter(writer)
-	defer tarWriter.Close()
+	tarWriter := tar.NewWriter(writer)
```

Comment:
I didn't try that, no! I can give it a go and push it with the other changes if that's the case.

Reply:
What about:
```
tarWriter := tar.NewWriter(writer)
err := makeTar(localPath, dest, writer, copyFiles, globExps, tarWriter)

other code bits...

tarWriter.Close()
```

Because creating a second tarWriter with the same writer didn't append the Dockerfile.
Would this be an adequate solution? It keeps the declaration/termination in the CopyFile function.